### PR TITLE
make selinux an (optional) feature

### LINKED ISSUE #1508: Make SELinux an (option) feature flag
**What would you like to be added**:
Make SELinux support optional as an feature flag. 

**Why is this needed**:
SELinux needs extended attributes filesystem, and as a building host, SELinux must be disabled so the security labels created during build times don't clash with active selinux policies.
We discovered that our build system has SELinux enabled by default with clashes with the `makepart` portion of the build system, which unconditionally set selinux attributes for the disk image files.

### PR DESCRIPTION
**Pull request by notandy**
**How to categorize this PR?**

/kind enhancement
/area os
/os garden-linux

**What this PR does / why we need it**:
This commit adds a feature flag `_selinux` that can be disabled for individual builds.
the `server` feature which is inherited by a few targets enables `_selinux` by default.
Disabling `_selinux` will also disable the selinux attribute tagging during partition build time.

This commit also cleanups the various hardcoded selinux `setfiles`.

**Which issue(s) this PR fixes**:
Fixes #1508

**Special notes for your reviewer**:

**Release note**:

```feature user

```

### CODE REVIEW

File: features/_selinux/test/test_kernel_cmdline.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+@pytest.mark.parametrize(
+    "file,args",
+    [
+        ("/etc/kernel/cmdline", "security=selinux"),
+    ]
+)
+
+
+def test_kernel_cmdline(client, file, args):
```
Comment (nkraetzschmar) [inline features/_selinux/test/test_kernel_cmdline.py:13]: please skip this test for githubActionRunner and firecracker feature by changing this to

```python
def test_kernel_cmdline(client, file, args, non_feature_githubActionRunner, non_feature_firecracker):
```

and adding a `non_feature_firecracker` pytest fixture in `tests/conftest.py`

```python
@pytest.fixture
def non_feature_firecracker(testconfig):
    features = testconfig.get("features", [])
    if "firecracker" in features:
        pytest.skip('test is not supported on firecracker')
```

File: features/firecracker/info.yaml [hunk 2]
```diff
@@ -3,3 +3,4 @@ type: platform
 features:
   include:
     - server
+  exclude: [ _selinux ]
```
Comment (nkraetzschmar) [inline features/firecracker/info.yaml:6]: exclude does not properly work yet, just skip the test instead

File: features/githubActionRunner/info.yaml [hunk 3]
```diff
@@ -1,2 +1,4 @@
 description: GitHub action self hosted runner
 type: element
+features:
+  exclude: [ _selinux ]
```
Comment (nkraetzschmar) [inline features/githubActionRunner/info.yaml:4]: see above

**Review by nkraetzschmar — CHANGES_REQUESTED** [review]

File: features/firecracker/info.yaml (hunk 2 above)
Comment (notandy) [reply features/firecracker/info.yaml:6]: Thanks, jeah just noticed it doesn't work

File: features/githubActionRunner/file.include/etc/kernel/cmdline.d/99-no-lsm.cfg [hunk 4]
```diff
@@ -1 +0,0 @@
-CMDLINE_LINUX="$(echo "$CMDLINE_LINUX" | tr ' ' '\n' | grep -v '^security=' | tr '\n' ' ' | sed -e 's/ $//')"
```
Comment (nkraetzschmar) [inline features/githubActionRunner/file.include/etc/kernel/cmdline.d/99-no-lsm.cfg:1]: please keep this file for now, until we get exclude features to work correctly

**Review by nkraetzschmar — CHANGES_REQUESTED** [review]

File: features/githubActionRunner/file.include/etc/kernel/cmdline.d/99-no-lsm.cfg (hunk 4 above)
Comment (notandy) [reply features/githubActionRunner/file.include/etc/kernel/cmdline.d/99-no-lsm.cfg:1]: true, forget to re-include it again. thanks.

**Review by nkraetzschmar — APPROVED** [review]

**Pull request merged**